File: 002compress.js
const compress = (s) => {
  // todo
  let result = [];
  let j = 0,
    i = 0;
  while (j < s.length) {
    if (s[j] === s[j + 1]) {
      j += 1;
    } else {
      const curSlice = s.slice(i, j + 1);
      if (curSlice.length === 1) {
        result.push(s[j]);
        j += 1;
        i = j;
      } else {
        result.push(curSlice.length + s[j]);
        j += 1;
        i = j;
      }
    }
  }
  return result.join("");
};

console.log(compress("ccaaatsss")); // -> '2c3at3s'

File: 002compress.py
def compress(s):
  s += '!'
  result = []
  j = 0
  i = 0

  while j < len(s):
    if s[i] == s[j]:
      j += 1
    else:
      num = j - i

      if num == 1:
        result.append( s[i] )
      else:
        result.append( str(num))
        result.append(s[i] )
        
      i = j

  return ''.join(result)
